src/iDealAdvancedConnector/Security/XsdValidation.cs

```diff
@@ -61,6 +61,9 @@ static XmlSchemaSet GetXsdSet(string[] xsdFileNames)
                 var schema = XmlSchema.Read(GetXsdFile(s), delegate { });
                 xsd.Add(schema);
             });
+
+            // compile xml schema set, to prevent the compilation is called again
+            xsd.Compile();
```

Comment:
maybe add as a comment that compilation is not thread safe, but validation is? (if that's the underlying issue as I understand it?)

Reply:
Added a comment

Reply:
perfect, thanks a lot!